apps/editing-toolkit/editing-toolkit-plugin/coming-soon/coming-soon.php

```diff
@@ -56,22 +56,44 @@ function coming_soon_page() {
 		return;
 	}
 
-	$id = (int) get_option( 'wpcom_coming_soon_page_id', 0 );
+	add_filter( 'wpcom_disable_logged_out_follow', '__return_true', 10, 1 ); // Disable follow actionbar.
+	add_filter( 'wpl_is_enabled_sitewide', '__return_false', 10, 1 ); // Disable likes.
+
+	$should_show_fallback = false;
 
+	$id = (int) get_option( 'wpcom_coming_soon_page_id', 0 );
 	if ( empty( $id ) ) {
-		render_fallback_coming_soon_page();
-		die();
+		$should_show_fallback = true;
 	}
 
-	if ( $post->ID === $id ) {
-		// We're already viewing the coming soon page, don't redirect.
-		return;
+	$custom_coming_soon_page = get_post( $id );
+	if ( ! $custom_coming_soon_page ) {
+		$should_show_fallback = true;
 	}
 
-	$coming_soon_page_url = get_page_link( $id );
-
-	if ( wp_safe_redirect( $coming_soon_page_url ) ) {
-		exit;
+	if ( $should_show_fallback ) {
+		render_fallback_coming_soon_page();
+		die();
 	}
+
+	?>
+	<!doctype html>
+	<html <?php language_attributes(); ?>>
+		<head>
+			<meta charset="<?php bloginfo( 'charset' ); ?>" />
+			<meta name="viewport" content="width=device-width, initial-scale=1" />
+			<?php wp_head(); ?>
+		</head>
+		<body>
+			<?php
+				// phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
```

Comment:
Is it okay to silence this line? Because we're just echoing out the post_content applied through `the_content`, is that safe enough?

Reply:
I copied what the `the_content()` function does. Doesn't fix the warning unfortunately, but I couldn't find a core escaping function that did the same thing as `str_replace( ']]>', ']]&gt;', ... )`